pstore/pstore.go

```diff
@@ -256,6 +256,10 @@ func (c *Consumer) flush() (err error) {
 			iter.Commit()
 		}
 	}
+	// Help out the GC
+	for i := range c.toBeCommitted {
+		c.toBeCommitted[i] = nil
+	}
 	// Clear out pending iterators
 	c.toBeCommitted = c.toBeCommitted[:0]
```

Comment:
How about simply creating a new slice (perhaps with the same capacity as the existing slice)? That seems cleaner and faster than looping over the slice elements and overwriting them.